sdk/communication/azure-communication-chat/src/main/java/com/azure/android/communication/chat/ChatClientBuilder.java

```diff
@@ -248,15 +218,17 @@ public ChatAsyncClient buildAsyncClient() {
             .endpoint(this.endpoint)
             .pipeline(pipeline);
 
-        return new ChatAsyncClient(clientBuilder.buildClient(), communicationSignalingClient);
+        return new ChatAsyncClient(clientBuilder.buildClient());
     }
 
     private HttpPipeline createHttpPipeline(HttpClient httpClient,
                                             HttpPipelinePolicy authorizationPolicy,
                                             List<HttpPipelinePolicy> additionalPolicies) {
 
         List<HttpPipelinePolicy> policies = new ArrayList<HttpPipelinePolicy>();
-        policies.add(authorizationPolicy);
+        if (authorizationPolicy != null) {
+            policies.add(authorizationPolicy);
+        }
```

Comment:
authorizationPolicy should be mandatory, should it not?

Reply:
Good question. This is the part I want to be reviewed.

There're 2 ways for customer to set authorizationPolicy: 1) add a HttpPipelinePolicy for it, 2) add a CommunicationTokenCredential and inside this client builder we create this authorizationPolicy from the credential.

In API review,  Srikanta suggested to remove function credentialPolicy() from client builder because we can use addPolicy() for it. If customer is using addPolicy() to set a HttpPipelinePolicy, it would be added to this.customPolicies. If customer is setting CommunicationTokenCredential, we would create this authorizationPolicy for him. So, authorizationPolicy would only present when customer is setting CommunicationTokenCredential. That's why it can be optional.

Besides, this means we lost the ability to validate whether customer has set a HttpPipelinePolicy or CommunicationTokenCredential for authorization. Would this be an issue?

Reply:
Same for ChatThreadClientBuilder

Reply:
We should validate that there is some mechanism for authorization or else we throw. I'm not a fan of using the generic `addPolicy()` to add the authorization policy, as it can make it hard to discover you need to provide an instance of `BearerTokenAuthenticationPolicy`. If we want to go this rout, we should be extremely clear in the JavaDoc about needing to authenticate the clients somehow and what options customers have. Builders should also throw when there's no form of authenticating when calling `buildClient()`/`buildAsyncClient()` and be super clear about what the problem is in the error message.

Reply:
I agree with Victor here. Chat client has to be clear on what kind of credential it supports. If we only support TokenCredential, we make that clear by throwing in the builder when it is null. We cannot make the promise that any kind of authentication policy would work. Therefore, assuming customer should somehow set an authentication policy is just not a safe assumption.